lua/plenary/path.lua

```diff
@@ -474,20 +474,77 @@ function Path:rename(opts)
   return status
 end
 
+--- Copy files or folders.
+---@param opts table: options to pass to toggling registered actions
+---@field destination string|Path: target file path to copy to
+---@field recursive bool: whether to copy folders recursively (default: true)
+---@field override bool: whether to override files (default: true)
+---@field interactive bool: confirm if copy would override; precedes `override` (default: false)
+---@field respect_gitignore bool: skip folders ignored by all detected `gitignore`s (default: true)
```

Comment:
does it make sense to have respect_gitignore to true? `cp` doesnt have it either.